user_sync/rules.py

```diff
@@ -447,6 +455,10 @@ def read_desired_user_groups(self, mappings, directory_connector_manager):
                                                            for umapi_name, umapi_info
                                                            in six.iteritems(self.umapi_info_by_name)]))
 
+    @staticmethod
+    def drop_user():
+        return bool
```

Comment:
This doesn't make sense as written - you're returning a type.  I think we can move drop_user inside the loop... let's talk on it